### DISCUSSION TITLE
References for ideas

### DISCUSSION THREAD

**Discussion by ccoVeille** [hougesen/mdsf#168] [Ideas]
The idea of this discussion is to use thread for random ideas that do not necessarily require an issue yet. But may need to discuss about them

**Comment by ccoVeille** [comment]
This is a rust highlighter, you may use it

https://github.com/trishume/syntect

Maybe in a verbose mode, as a diff or as before/after provided with highlighting in the console

**Comment by ccoVeille** [comment]
By creating #161 I found this interesting resources:

https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#syntax-highlighting

I think it's the list of "fenced blockquote" known by GitHub Flavored Markdown

https://github.com/github-linguist/linguist/blob/master/vendor/README.md

This file comes with interesting information such as grammar parser

https://github.com/github-linguist/linguist/blob/master/lib/linguist/languages.yml

**Reply by ccoVeille** [reply]
It's fun to look at the https://gh.io/acemodes then as mentionned in the header.

you will find the js code used to highlight them, so you can see their format

**Reply by ccoVeille** [reply]
so if you support shell, you can add zsh, bash, shell-script

You might not need this now but it's interesting

https://github.com/github-linguist/linguist/blob/master/lib/linguist/languages.yml#L6533

EDIT: This idea led to alternative shell formatting with #164

**Comment by ccoVeille** [comment]
Other source of inspiration

https://github.com/benwis/femark

I'm curious how they handled the go partial code if they did

**Comment by ccoVeille** [comment]
This article explains what GitHub does for syntax highlighting via linguist

https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#syntax-highlighting

Mdsf is not about highlighting. But it could lead to interesting thing if you could use linguist or something like that to identify:

- language in a fenced block without language hint => so you could fix them

````markdown
```
{"value"}
```
````

````markdown
```json
{"value"}
```
````

- fenced block using incorrect language

````markdown
```json
{"value"} // comment
```
````

````markdown
```jsonc
{"value"} // comment
```
````

**Comment by ccoVeille** [comment]
Found

https://github.com/bitst0rm-pub/Formatter

It lists a lot of plugins and linter/formatter, some may be useful for your project

**Reply by hougesen** [reply]
I just checked the list and pretty much everything is already supported 😆

**Reply by hougesen** [reply]
Only ones missing is the R formatters, but they are both a pain to work with since they require us to start and R repl (IIRC)
